### DISCUSSION TITLE
Always display validation rules even if there are no errors

### DISCUSSION THREAD

**Discussion by comhon-project** [formkit/formkit#851] [Ideas]
idea:
it would be nice to have an option to always show the validation rules. 
for example, in a list, I would like users to know that they can specify a maximum of three items before specifying the fourth item.

proposal : 
define a new visibility, something like `validation-visibility="permanent"`

**Comment by justin-schroeder** [comment]
Interesting. Wouldn’t using help text be better for this? I would be concerned about how the actual languages of the messages are written would cause users to be very confused.

**Comment by fenilli** [comment]
The problem I can see is mostly that messages is a validation/error thing already, so if you add that those would already be visible at the start, not only your message.

Only thing I can think of that could be interesting in this line of thinking is if messages could tell about the visibility state themselves `{ key: 'help', value: 'can only do three', visibility: 'permanent' }` and each one would be able to tell when it is able to show.

Because there isn't a validation error to begin with, so why would you show a message ( this is like @justin-schroeder said the helps job ).

**Comment by comhon-project** [comment]
@justin-schroeder  @GustavoFenilli using help is probably a better solution. But I don't want to redefine all messages for all types of validations myself, I would like to re-use existing messages. 
So maybe another solution would be to tell formkit input to display all validations rules in the help container (or a new dedicated container?).

something like : 
```
<Formkit type="number" validation="max:3" :show-validation-rules="true" />
```

**Comment by fenilli** [comment]
What would happen to those messages when there isn't a validation error? and isn't the `live` what you want? it [shows validations](https://formkit.link/40d99172789c0d400e457523992088c5) at start.

**Reply by comhon-project** [reply]
@GustavoFenilli the purpose is to display rules even if the input is valid, so I can't use `live` visibility. 
As said in my first comment, for example I would like to inform user (from the start) that he can add at most 3 items. to avoid him to add a fourth item (that will display error message) and remove it to respect the rule.
And i want to re-use validation messages to avoid to redefine my own messages (for each rule and each locale)  in the help section.

**Comment by comhon-project** [comment]
@justin-schroeder @GustavoFenilli I'm not sure I explain well... does it make sens for you ? is it worth it ?

**Reply by fenilli** [reply]
As for validation I don't think so, its counter intuitive that a validation is still being show even if there isn't a validation error.

It might be possible to do something in these lines with a plugin by getting the validation messages for the validations and adding to the message store with visibility and non blocking.

**Reply by justin-schroeder** [reply]
Technically you could write a plugin that used the `node.parsedValidationRules` and then looked up the i18n message for each of them and set a non-blocking message with the same value. I really think this is the wrong approach though. I would highly recommend using help text for this 👍
